lib/src/arg_parser.dart

```diff
@@ -134,6 +134,7 @@ class ArgParser {
       bool negatable = true,
       void Function(bool)? callback,
       bool hide = false,
+      bool ignoreMandatory = false,
```

Comment:
We need to document this on the method as well.

I would suggest some wording like:

```
If [ignoreMandatory] is `true`, then passing any explicit value for this flag will cause
all other mandatory options to not be mandatory. 
```

This brings up another question though, it is pretty weird to me that this takes effect whenever the flag is passed - regardless of the value passed for it.

In the case of `--help` and `--verbose`, we would want this to only happen if the value was `true` for instance.

Reply:
Thanks for the documentation, I had forgotten it.

I can add a new check to make sure, if there is a value passed to the flag, that it's `true`.

Reply:
> I can add a new check to make sure, if there is a value passed to the flag, that it's `true`.

Ya, I do somewhat wonder if we should expose the ability to configure which value (true or false) should trigger it?

For instance we could change it to `bool? ignoreMandatoryIf`, and then you would either not set it at all, or pass an explicit boolean value which is the state you want to use to ignore mandatory options.

For instance now with the `true` check there is yet another bit of weirdness - what if you set `ignoreMandatory` on a flag whose default value is `true`? Now explicitly enabling it disables the checks but they aren't disabled by default, even though you haven't actually changed the values of any flags.

Reply:
Another option could be to make `defaultsTo: true` and `ignoreMandatory: true` be incomatible

Reply:
> 
> 
> Another option could be to make `defaultsTo: true` and `ignoreMandatory: true` be incompatible

Yeah, this is definitively not two compatible options.